src/register/register.ts

```diff
@@ -302,9 +302,7 @@ export class Register {
    */
   private static processNumberedRegister(content: RegisterContent, vimState: VimState): void {
     // Find the BaseOperator of the current actions
-    const baseOperator = vimState.recordedState.actionsRun.find(value => {
-      return value instanceof BaseOperator || value instanceof BaseCommand;
-    });
```

Comment:
Previously, this code was initializing `baseOperator` to be `CommandNumber` in some cases. For example, `5Y` is `[CommandNumber, CommandYankFullLine]`, and because [`CommandNumber` inherits from `BaseCommand`](https://github.com/VSCodeVim/Vim/blob/bd153ad9a8e7b3a445ebcbd596d7eeaf994e92a4/src/actions/commands/actions.ts#L284) and comes first, `baseOperator` would be set to that.

Then, the blocks below specifically to handle yanks wouldn't be hit.

https://github.com/VSCodeVim/Vim/blob/bd153ad9a8e7b3a445ebcbd596d7eeaf994e92a4/src/actions/commands/actions.ts#L284